esphome/components/ld2420/number/__init__.py

```diff
@@ -0,0 +1,183 @@
+)
+
+
+async def to_code(config):
+    LD2420_component = await cg.get_variable(config[CONF_LD2420_ID])
+    if gate_timeout_config := config.get(CONF_PRESENCE_TIMEOUT):
+        n = await number.new_number(
+            gate_timeout_config, min_value=0, max_value=255, step=5
+        )
+        await cg.register_parented(n, config[CONF_LD2420_ID])
+        cg.add(LD2420_component.set_gate_timeout_number(n))
+    if min_distance_gate_config := config.get(CONF_MIN_GATE_DISTANCE):
+        n = await number.new_number(
+            min_distance_gate_config, min_value=0, max_value=15, step=1
+        )
+        await cg.register_parented(n, config[CONF_LD2420_ID])
+        cg.add(LD2420_component.set_min_gate_distance_number(n))
+    if max_distance_gate_config := config.get(CONF_MAX_GATE_DISTANCE):
+        n = await number.new_number(
+            max_distance_gate_config, min_value=1, max_value=15, step=1
+        )
+        await cg.register_parented(n, config[CONF_LD2420_ID])
+        cg.add(LD2420_component.set_max_gate_distance_number(n))
+    if gate_move_sensitivity_config := config.get(CONF_GATE_MOVE_SENSITIVITY):
+        n = await number.new_number(
+            gate_move_sensitivity_config, min_value=0.05, max_value=1, step=0.025
+        )
+        await cg.register_parented(n, config[CONF_LD2420_ID])
+        cg.add(LD2420_component.set_gate_move_sensitivity_factor_number(n))
+    if gate_still_sensitivity_config := config.get(CONF_GATE_STILL_SENSITIVITY):
+        n = await number.new_number(
+            gate_still_sensitivity_config, min_value=0.05, max_value=1, step=0.025
+        )
+        await cg.register_parented(n, config[CONF_LD2420_ID])
+        cg.add(LD2420_component.set_gate_still_sensitivity_factor_number(n))
+    if config.get(CONF_GATE_SELECT):
+        if gate_number := config.get(CONF_GATE_SELECT):
+            n = await number.new_number(gate_number, min_value=0, max_value=15, step=1)
+            await cg.register_parented(n, config[CONF_LD2420_ID])
+            cg.add(LD2420_component.set_gate_select_number(n))
+        if gate_still_threshold := config.get(CONF_STILL_THRESHOLD):
+            n = cg.new_Pvariable(gate_still_threshold[CONF_ID])
+            await number.register_number(
+                n, gate_still_threshold, min_value=0, max_value=65535, step=25
+            )
+            await cg.register_parented(n, config[CONF_LD2420_ID])
+            cg.add(LD2420_component.set_gate_still_threshold_numbers(0, n))
+        if gate_move_threshold := config.get(CONF_MOVE_THRESHOLD):
+            n = cg.new_Pvariable(gate_move_threshold[CONF_ID])
+            await number.register_number(
+                n, gate_move_threshold, min_value=0, max_value=65535, step=25
+            )
+            await cg.register_parented(n, config[CONF_LD2420_ID])
+            cg.add(LD2420_component.set_gate_move_threshold_numbers(0, n))
+    else:
+        for x in range(16):
+            if gate_conf := config.get(f"g{x}"):
```

Comment:
Ah, found the issue to [this comment](https://github.com/esphome/esphome/pull/4847#issuecomment-1790097404):
```suggestion
            if gate_conf := config.get(f"gate_{x}"):
```

As and alternative, make this prefix a constant somewhere above. haha

Reply:
Added.